api/environments/identities/managers.py

```diff
@@ -30,3 +30,11 @@ def get_or_create_for_sdk(
             .prefetch_related("identity_traits")
             .get_or_create(identifier=identifier, environment=environment)
         )
+
+    def with_overrides(self) -> QuerySet["Identity"]:
+        """
+        Only include identities with overridden features.
+        """
+        return self.annotate(overridden_count=Count("identity_features")).filter(
+            overridden_count__gt=0
+        )
```

Comment:
I'm not sure it's the best way to do this, given that `filter_for_environment_document` prefetches all the feature states.

Perhaps this could be delegated to pure Python in the mapper by iterating `environment.feature_states.all()`.

Reply:
This might also get painful in the EE merge so I'm certainly for doing this in pure python.

Reply:
Couldn't avoid it but the SQL is much more straightforward now.